datalad/interface/base.py

```diff
@@ -329,24 +330,25 @@ def _prep(
 
         Paths are then sorted by the datasets that contain them. If paths are
         detected that are not associated with any dataset `ValueError` is
-        raised.
+        raised. If a `dataset` is given, any paths associated with a dataset
+        that is not this dataset or a subdataset of it will also trigger a
+        `ValueError`.
```

Comment:
I take that back. This exception needs knowledge which repo we are taking about. I don't know that in all cases. Just that it is none. At the same time I want the same exception in all cases.